service/cluster/cluster.go

```diff
@@ -98,6 +99,7 @@ func New(opts Options) (*Service, error) {
 		AdvertiseAddr: opts.AdvertiseAddress,
 		Log:           l,
 		Sharder:       shard.Ring(tokensPerNode),
+		Label:         opts.ClusterName,
```

Comment:
One more thing to point out: in case we use this label with the current [mixin alerts](https://github.com/grafana/agent/blob/e966a94511e0dab32137227bebd54b24c9c35787/operations/agent-flow-mixin/alerts/clustering.libsonnet#L9) and deploy two different Agent clusters on the same K8S cluster and namespace, it would always trigger `ClusterNotConverging`.

Is there a way to avoid this by passing ClusterName as an extra label on metrics? I'm not even sure how the query would work when a cluster has a label but another doesn't (eg. the behavior of `by` clauses for non-existing labels).

It's not think it's necessarily a blocker for now, but we should at least investigate and open an issue to track and fix before the next release.

Reply:
can't we just ignore the clusterName the same way we ignore the state in the query?